test/components/views/dialogs/SpotlightDialog-test.tsx

```diff
@@ -214,200 +206,165 @@ describe("Spotlight Dialog", () => {
-            await act(async () => {
-                await sleep(200);
-            });
-            wrapper.update();
+            render(<SpotlightDialog initialText={testPerson.display_name} onFinished={() => null} />);
+            jest.advanceTimersByTime(200);
+            await flushPromisesWithFakeTimers();
 
-            const filterChip = wrapper.find("div.mx_SpotlightDialog_filter");
-            expect(filterChip.exists()).toBeTruthy();
-            expect(filterChip.text()).toEqual("People");
+            fireEvent.click(screen.getByText("People"));
 
-            const content = wrapper.find("#mx_SpotlightDialog_content");
-            const options = content.find("div.mx_SpotlightDialog_option");
-            expect(options.length).toBeGreaterThanOrEqual(1);
-            expect(options.first().text()).toContain(testPerson.display_name);
+            // search is debounced
+            jest.advanceTimersByTime(200);
+            await flushPromisesWithFakeTimers();
 
-            wrapper.unmount();
+            const filterChip = document.querySelector("div.mx_SpotlightDialog_filter")!;
+            expect(filterChip).toBeInTheDocument();
+            expect(filterChip.innerHTML).toContain("People");
+
+            const content = document.querySelector("#mx_SpotlightDialog_content")!;
+            const options = content.querySelectorAll("div.mx_SpotlightDialog_option");
+            expect(options.length).toBeGreaterThanOrEqual(1);
+            expect(options[0]!.innerHTML).toContain(testPerson.display_name);
         });
     });
 
     describe("should allow clearing filter manually", () => {
         it("with public room filter", async () => {
-            const wrapper = mount(<SpotlightDialog initialFilter={Filter.PublicRooms} onFinished={() => null} />);
-            await act(async () => {
-                await sleep(200);
-            });
-            wrapper.update();
-
-            let filterChip = wrapper.find("div.mx_SpotlightDialog_filter");
-            expect(filterChip.exists()).toBeTruthy();
-            expect(filterChip.text()).toEqual("Public rooms");
+            render(<SpotlightDialog initialFilter={Filter.PublicRooms} onFinished={() => null} />);
+            // search is debounced
+            jest.advanceTimersByTime(200);
+            await flushPromisesWithFakeTimers();
 
-            filterChip.find("div.mx_SpotlightDialog_filter--close").simulate("click");
-            await act(async () => {
-                await sleep(1);
-            });
-            wrapper.update();
+            let filterChip = document.querySelector("div.mx_SpotlightDialog_filter")!;
+            expect(filterChip).toBeInTheDocument();
+            expect(filterChip.innerHTML).toContain("Public rooms");
 
-            filterChip = wrapper.find("div.mx_SpotlightDialog_filter");
-            expect(filterChip.exists()).toBeFalsy();
+            fireEvent.click(filterChip.querySelector("div.mx_SpotlightDialog_filter--close")!);
```

Comment:
The close button has an alt-text (`Remove search filter for Public rooms`) that we might use.